tests/ProgramsTest.hs

```diff
@@ -50,16 +52,24 @@ do_test = do
 
   withSystemTempDirectory "c-h-programs-test" $ \tmpdir -> do
 
+  let ext = case System.Info.os of
+              "mingw32" -> "exe"
+              _         -> ""
+
   forM_ ["8.6.5", "8.4.4"] $ \ver -> do
 
-  let ghc = tmpdir </> "ghc-" ++ ver
-  let ghc_pkg = tmpdir </> "ghc-pkg-" ++ ver
-  let haddock = tmpdir </> "haddock-" ++ ver
+  let ghc = tmpdir </> "ghc-" ++ ver <.> ext
+  let ghc_pkg = tmpdir </> "ghc-pkg-" ++ ver <.> ext
+  let haddock = tmpdir </> "haddock-" ++ ver <.> ext
   let progs = defaultPrograms { ghcProgram = ghc }
 
-  createSymbolicLink prog ghc
-  createSymbolicLink prog ghc_pkg
-  createSymbolicLink prog haddock
+  let link = case System.Info.os of
+              "mingw32" -> copyFile
```

Comment:
Why do you need this, AFAIK symlinks are usable on Windos now, no?

Reply:
TL;DR: symbolic links are not well supported by haskell + windows

Well, it comes from `System.Posix.Files` via a reexport in `Symlink`. I think it simply doesn't work in windows. In fact it throws an error of `exe not found` in my windows 10.

We could change it to `System.Posix.Compat`, buuuut it simply throws an error afaiu:
https://github.com/jacobstanley/unix-compat/blob/master/src/System/PosixCompat/Files.hsc#L385

Again we could use  [`System.Win32.SymbolicLink.createSymbolicLink`](https://hackage.haskell.org/package/Win32-2.8.5.0/docs/System-Win32-SymbolicLink.html#v:createSymbolicLink) but i think there is a good reason for unix-compat to not use it:
```
Note: You should worry about UAC (User Account Control) when use this module's function in your application:

    require to use 'Run As Administrator' to run your application.
    or modify your application's manifect file to add <requestedExecutionLevel level=requireAdministrator uiAccess=false/>.
```

Reply:
I'm not sure I understand what's going on here. Can you try doing a quick test with `Symlink.createSymbolicLink` directy to see why it's trowing an error? Is it because we're symlinking an exe? Is it because the symlink target doesn't exist? (It should, right?)

I just don't like papering over problems we don't understand.

Reply:
I was wrong about what module is using c-h: it is using `System.Win32.SymbolicLink` (the second one i mentioned) using a coditional in  the `.cabal` file.

But i am not able to make it work. Maybe i am using it wrong, consider this repl session:
```
PS D:\dev\ws\haskell\cabal-helper> cabal repl
Build profile: -w ghc-8.8.3 -O1
In order, the following will be built (use -v for more details):
 - cabal-helper-1.1.0.0 (lib) (first run)
Preprocessing library for cabal-helper-1.1.0.0..
GHCi, version 8.8.3: https://www.haskell.org/ghc/  :? for help
Ok, 24 modules loaded.
Prelude Distribution.Helper> import Symlink
Prelude Symlink Distribution.Helper> :! D:\bin\happy.exe -v
Happy Version 1.19.12 Copyright (c) 1993-1996 Andy Gill, Simon Marlow (c) 1997-2005
Prelude Symlink Distribution.Helper> createSymbolicLink "D:\\bin\\happy.exe" ".\\happy2.exe"
Prelude Symlink Distribution.Helper> :! dir .\happy* /s
 El volumen de la unidad D es DATA
No se encuentra el archivo
```

I've tested directly  `System.Win32.SymbolicLink.createSymbolicLinkFile` with the same result.
🤔

Reply:
So here you're not even getting an error?! createSymbolicLink is just a nop? WTF?

I would probably go figure out what winapi function is being used and write a little C program to test it directly..

Reply:
Can you try with a normal file too?

Reply:
Well i should have posted here or in an issue the error i got before replacing symlink with copy and made me do it:

```
PS D:\dev\ws\haskell\cabal-helper> git diff
diff --git a/tests/ProgramsTest.hs b/tests/ProgramsTest.hs
index 1c28ced..f8285d2 100644
--- a/tests/ProgramsTest.hs
+++ b/tests/ProgramsTest.hs
@@ -60,9 +60,10 @@ do_test = do
   let haddock = tmpdir </> "haddock-" ++ ver <.> exeExtension

-  let link = case System.Info.os of
-              "mingw32" -> copyFile
-              _         -> createSymbolicLink
+  let link = createSymbolicLink
+             -- case System.Info.os of
+             -- "mingw32" -> copyFile
+             -- _         -> createSymbolicLink

PS D:\dev\ws\haskell\cabal-helper> cabal test programs-test
Build profile: -w ghc-8.8.3 -O1
In order, the following will be built (use -v for more details):
..... build ouput
Test suite programs-test: RUNNING...
Searching for C:\TEMP\c-h-programs-test15996\ghc-8.6.5.exe in path.
Cannot find C:\TEMP\c-h-programs-test15996\ghc-8.6.5.exe on the path
programs-test.exe: dieVerbatim: user error (CallStack (from HasCallStack):
  die', called at libraries\Cabal\Cabal\Distribution\Simple\Program\Db.hs:335:25 in Cabal-3.0.1.0:Distribution.Simple.Program.Db
  configureProgram, called at libraries\Cabal\Cabal\Distribution\Simple\Program\Db.hs:460:16 in Cabal-3.0.1.0:Distribution.Simple.Program.Db
  lookupProgramVersion, called at libraries\Cabal\Cabal\Distribution\Simple\Program\Db.hs:498:3 in Cabal-3.0.1.0:Distribution.Simple.Program.Db
  requireProgramVersion, called at libraries\\Cabal\\Cabal\\Distribution\\Simple\\GHC.hs:136:5 in Cabal-3.0.1.0:Distribution.Simple.GHC
  configure, called at src\CabalHelper\Compiletime\CompPrograms.hs:35:39 in main:CabalHelper.Compiletime.CompPrograms
programs-test.exe: Cannot find the program 'ghc'. User-specified path
'C:\TEMP\c-h-programs-test15996\ghc-8.6.5.exe' does not refer to an executable
and the program is not on the system path.
)
Test suite programs-test: FAIL
Test suite logged to:
D:\dev\ws\haskell\cabal-helper\dist-newstyle\build\x86_64-windows\ghc-8.8.3\cabal-helper-1.1.0.0\t\programs-test\test\cabal-helper-1.1.0.0-programs-test.log
0 of 1 test suites (0 of 1 test cases) passed.
cabal.exe: Tests failed for test:programs-test from cabal-helper-1.1.0.0.
```

Reply:
I would conclude that if you dont run the program as administrator or add the config as suggested in the module the function simply does nothing 😞

Reply:
I think it should throw an error if the app has not enough privileges to execute it. I guess it rely in the win32 api behaviour and check if the running program has privileges to do it before do the actual call would be tricky or impose a performance penalty.
But at least the docs should war about this.